test/test_charm.py

```diff
@@ -128,6 +128,30 @@ def _on_start(self, event: ops.EventBase):
         # check that the event has been seen by the observer
         self.assertIsInstance(charm.seen, ops.StartEvent)
 
+    def test_observe_nested_object_noninstantiated(self):
```

Comment:
```suggestion
    def test_observer_not_referenced_warning(self):
```

Also, should this test live in `test_framework.py`, as the relevant code lives in `framework.py`?

Reply:
I thought about it, but it felt way too complicated to implement it without referring to the charm object. If you look at the test immediately above it, I feel it makes sense to keep it here as they're very similar.
If you really want that that's a significant rework of the test.